core/components/atoms/select/select.js

```diff
@@ -12,6 +13,22 @@ const selectOpacity = {
 }
 const PLACEHOLDER_VALUE = '__select_placeholder'
 
+const StyledIcon = styled(Icon).attrs({
+  name: 'dropdown-fill',
+  size: '14'
+})`
```

Comment:
Instead of extending the Icon component, can we use this as props in the place where it is used?

```jsx
<Icon name="dropdown-fill" size="14" />
```

and then add styles in the wrapper?

```js
Select.Wrapper = styled.div`
  ${Icon} {
    right: 12px;
  }
`
```

Whenever possible, we should prefer React API over styled-components API

### Why?

- Single consistent way of using components
- Makes it easier to upgrade (or replace) styled-components
- Set the right example for how we want folks to use cosmos components by using them right inside the system as well

Reply:
Completely agree with you, just fixed it :)